brainio/lookup.py

```diff
@@ -54,36 +55,51 @@ def data():
 
 def list_stimulus_sets():
     stimuli_rows = data()[data()['lookup_type'] == TYPE_STIMULUS_SET]
-    return list(set(stimuli_rows['identifier']))
+    return sorted(list(set(stimuli_rows['identifier'])))
 
 
 def list_assemblies():
     assembly_rows = data()[data()['lookup_type'] == TYPE_ASSEMBLY]
-    return list(assembly_rows['identifier'])
+    return sorted(list(set(assembly_rows['identifier'])))
 
 
 def lookup_stimulus_set(identifier):
     lookup = data()[(data()['identifier'] == identifier) & (data()['lookup_type'] == TYPE_STIMULUS_SET)]
     if len(lookup) == 0:
         raise StimulusSetLookupError(f"stimulus_set {identifier} not found")
-    if len(lookup) > 2:
-        raise RuntimeError(
-            f"Internal data inconsistency: Found more than 2 lookup rows for stimulus_set identifier {identifier}")
-    csv_lookup = [lookup_row for _, lookup_row in lookup.iterrows() if _is_csv_lookup(lookup_row)]
-    zip_lookup = [lookup_row for _, lookup_row in lookup.iterrows() if _is_zip_lookup(lookup_row)]
-    assert len(csv_lookup) == 1 and len(zip_lookup) == 1
-    csv_lookup, zip_lookup = csv_lookup[0], zip_lookup[0]
+    csv_lookup = _lookup_stimulus_set_filtered(lookup, _is_csv_lookup, "CSV")
+    zip_lookup = _lookup_stimulus_set_filtered(lookup, _is_zip_lookup, "zip")
     return csv_lookup, zip_lookup
 
 
+def _lookup_stimulus_set_filtered(lookup, filter_func, label):
```

Comment:
I don't understand the purpose of the label here, it only seems to be used in error messages but not in actual retrieval?

Reply:
It's just so the error is useful.

Reply:
still confused: where are the `_is_csv_lookup` and `_is_zip_lookup` variables even defined? It feels like something is missing here, and likely could be simplified. Passing an error label feels wrong to me

Reply:
They're module-level functions.  You wrote them.  https://github.com/brain-score/brainio_collection/blame/5047b8b2eb8311bfc5563ecd1420134151eeed02/brainio_collection/lookup.py#L90 

My change was likely a compulsive attempt at DRY combined with trying to change as little as possible.  There's a big block of behavior that happens twice, and the part that varies is in the middle.

Reply:
ooh I see now, I didn't catch that these were functions. Could you pass them in as kwargs, I think that would have helped me: `_lookup_stimulus_set_filtered(lookup, filter_func_is_csv_lookup, label="CSV")`

Reply:
Good idea.